vscode-extension/src/providers/register.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import * as semver from 'semver';
+import {workspace} from 'vscode';
+import {RelayExtensionContext} from '../context';
+import {RelayTextDocumentContentProvider} from './textDocumentContentProvider';
+
+export function registerProviders(context: RelayExtensionContext) {
+  const {binaryVersion} = context.relayBinaryExecutionOptions;
+
+  if (binaryVersion) {
+    const hasConfigJsonSchemaCommand =
+      semver.satisfies(binaryVersion, '>=18.0') ||
```

Comment:
This probably needs to be updated depending on what the release schedule will be like. Just picked the next major for now.

Reply:
Since it will be in the next release, can we just say `>17.0.0`?

Reply:
Sorry, one more realization. What happens if you use a version of the compiler from the @main NPM tag? Will that work or not? Would be cool to be able to test this end to end before the next release by installing the new version of the extension and installing relay-compiler@main (note that @main releases are currently stuck due to broken CI)